src/routes/auth/controllers/githubCallback.ts

```diff
@@ -9,10 +9,15 @@ const tokenExpiration: number = 3 * HOUR;
 
 export default async (ctx: Context, next: Next): Promise<void> => {
   const accessCode: string = ctx.query.code;
-  const newUser: UserDocument | null = await processGithubOAuth(accessCode);
+  const newUser: UserDocument = (await processGithubOAuth(accessCode)) as UserDocument;
   const jwtToken: string | undefined = newUser ? getToken(newUser, tokenExpiration) : undefined;
   if (jwtToken) {
-    ctx.cookies.set('token', jwtToken, { httpOnly: true, maxAge: tokenExpiration });
+    // ctx.cookies.set('nickname', newUser.nickname, { httpOnly: false, maxAge: tokenExpiration });
+    ctx.cookies.set('token', jwtToken, { httpOnly: false, maxAge: tokenExpiration });
+    /**
+     * @TODO
+     * http only option
+     */
   }
```

Comment:
newUser가 UserDocument 타입으로 보장된다면, getToken 함수 반환값을 string으로 고정하고 jwtToken의 타입에서도 undefined는 빼주어도 될 것 같습니다! 😁 (processGithubOAuth 함수 쪽에서 DB 요청 비동기 에러가 나면 에러핸들러로 보내주어서 newUser 값이 보장된다면 더욱 좋을 것 같네요)
```suggestion
  const newUser: UserDocument = (await processGithubOAuth(accessCode)) as UserDocument;
  const jwtToken: string = getToken(newUser, tokenExpiration);
  ctx.cookies.set('nickname', newUser.nickname, { httpOnly: false, maxAge: tokenExpiration });
  ctx.cookies.set('token', jwtToken, { httpOnly: false, maxAge: tokenExpiration });
  /**
   * @TODO
   * http only option
   */
```